processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -651,9 +650,10 @@ public ResultRow apply(Object input)
             );
 
             if (isResultLevelCache) {
-              Iterator<PostAggregator> postItr = query.getPostAggregatorSpecs().iterator();
-              int postPos = 0;
-              while (postItr.hasNext() && results.hasNext()) {
+              for (int postPos = 0; postPos < query.getPostAggregatorSpecs().size(); postPos++) {
+                if (!results.hasNext()) {
+                  throw new ISE("Ran out of objects while reading postaggs from cache!");
```

Comment:
can use DruidException.defensive instead.

Reply:
I've changed it to `DruidException` - I was just following the error handling which was already in this method; there is another `ISE` a few lines below and `fetchAggregatorsFromCache` also throws an `ISE`